packages/sn-controls-react/src/viewcontrols/EditView.tsx

```diff
@@ -41,89 +30,66 @@ export interface EditViewState {
  *  <EditView content={selectedContent} onSubmit={editSubmitClick} />
  * ```
  */
-export class EditView extends Component<EditViewProps, EditViewState> {
-  constructor(props: any) {
-    super(props)
-    const controlMapper = reactControlMapper(this.props.repository)
-    this.state = {
-      content: this.props.content,
-      schema: controlMapper.getFullSchemaForContentType(this.props.content.Type, 'edit'),
-      saveableContent: {} as ContentType,
-      controlMapper,
-    }
+export const EditView: React.FC<EditViewProps> = props => {
+  const controlMapper = reactControlMapper(props.repository)
+  const schema = controlMapper.getFullSchemaForContentType(props.content.Type, 'edit')
+  const [saveableFields, setSaveableFields] = useState({})
 
-    this.handleInputChange = this.handleInputChange.bind(this)
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    props.onSubmit && props.onSubmit(props.content, saveableFields as any)
```

Comment:
```suggestion
    props.onSubmit?.(props.content, saveableFields as any)
```
With the new TypeScript 3.7 we no longer need to write code like this: `object.function && object.function()`
we can simplify this with this syntax `object.function?.()` . This will not execute the function if it is null or undefined.